examples/energy_mng/src/controllers/weather.rs

```diff
@@ -0,0 +1,131 @@
+// Copyright © SixtyFPS GmbH <info@slint-ui.com>
+// SPDX-License-Identifier: GPL-3.0-only OR LicenseRef-Slint-commercial
+
+use crate::ui::*;
+use chrono::prelude::*;
+use core::cmp::Ordering;
+use slint::*;
+use std::rc::Rc;
+use weer_api::*;
+
+const WEATHER_API_KEY: &str = "WEATHER_API";
+const LAT_BERLIN: f32 = 52.520008;
+const LONG_BERLIN: f32 = 13.404954;
+const FORECAST_DAYS: i64 = 5;
+
+pub fn setup(window: &MainWindow) {
+    if let Some(api_key) = std::env::var_os(WEATHER_API_KEY) {
+        if let Some(api_key) = api_key.to_str() {
+            let now = Local::now();
+            let mut day_counter = 1;
+            let mut forecast_days = vec![];
+
+            let client = Client::new(api_key, true);
+
+            loop {
+                let forecast_date = now + chrono::Duration::days(day_counter);
+
+                if let Ok(forecast) = client
+                    .forecast()
+                    .query(Query::Coords(LAT_BERLIN, LONG_BERLIN))
+                    .dt(forecast_date)
+                    .call()
```

Comment:
Is that a synchronous API?
We should call that in a thread ideally.
And also do all the calls in parallel.

Reply:
This is synchronous. But yes it can be done in a separate thread.